agent/src/agent.php

```diff
@@ -131,10 +135,10 @@
     maxBufferDurationInSeconds: $debug ? 1 : 10,
     onIngestSuccess: static fn (ResponseInterface $response, float $duration) => $info('Ingest successful ['.round($duration, 3).'s]'),
     onIngestError: static fn (Throwable $e, float $duration) => $info('Ingest failed ['.round($duration, 3).'s]: '.$e->getMessage()),
-    onExceededQuota: static function (float $duration) use ($info, $ingestDetails) {
+    onOverQuota: static function (float $duration) use ($info, $ingestDetails) {
         $info('Ingest attempted ['.round($duration, 3).'s]: Quota exceeded');
 
-        $ingestDetails->markQuotaExceeded();
+        $ingestDetails->markOverQuota();
```

Comment:
What do you think about calling `$ingestDetails->markOverQuota();` within the `Ingest` class instead of calling it via the callback, as it already has access to `$ingestDetails`?

Reply:
I started writing up my reasoning for why I went with this approach and realised it didn't really hold up.

I've moved this into the Ingest as you suggested.

Really appreciate this and the other comments.